src/main/java/org/fcrepo/transform/transformations/LDPathTransform.java

```diff
@@ -78,60 +74,57 @@ public LDPathTransform(final InputStream query) {
 
-        final Set<NodeType> supertypes = orderedBy(BY_NAME).add(primaryNodeType.getSupertypes()).build();
-        final Set<NodeType> mixinTypes = orderedBy(BY_NAME).add(node.getMixinNodeTypes()).build();
+        final List<URI> rdfTypes = resource.getTypes();
 
-        // start with mixins, primary type, and supertypes of primary type
-        final ImmutableList.Builder<NodeType> nodeTypesB = builder();
-        nodeTypesB.addAll(mixinTypes).add(primaryNodeType).addAll(supertypes);
+        LOGGER.debug("Discovered rdf types: {}", rdfTypes);
 
-        // add supertypes of mixins
-        mixinTypes.stream().map(mixin -> orderedBy(BY_NAME).add(mixin.getDeclaredSupertypes()).build())
-            .forEach(nodeTypesB::addAll);
+        final NamespaceRegistry nsRegistry = resource.getNode().getSession().getWorkspace().getNamespaceRegistry();
 
-        final List<NodeType> nodeTypes = nodeTypesB.build();
-
-        LOGGER.debug("Discovered node types: {}", nodeTypes);
-        for (final NodeType nodeType : nodeTypes) {
-            if (programNode.hasNode(nodeType.toString())) {
-                return new LDPathTransform(programNode.getNode(nodeType.toString())
-                                               .getNode(JCR_CONTENT)
-                                               .getProperty(JCR_DATA)
-                                               .getBinary().getStream());
+        // convert rdf:type with URI namespace to prefixed namespace
+        final Function<URI, String> namespaceUriToPrefix = x -> {
+            final String uriString = x.toString();
+            try {
+                for (final String namespace : nsRegistry.getURIs()) {
+                    // Ignoring zero-length namespaces return the appropriate prefix
+                    if (namespace.length() > 0 && uriString.startsWith(namespace)) {
+                        return uriString.replace(namespace, nsRegistry.getPrefix(namespace) + ":");
+                    }
+                }
+                return uriString;
+            } catch (final RepositoryException e) {
+                return uriString;
             }
-        }
+        };
+
+        final List<String> rdfStringTypes = rdfTypes.stream().map(namespaceUriToPrefix)
+                .map(stringType -> transformResource.getPath() + "/" + stringType + "/jcr:content")
+                .collect(Collectors.toList());
 
-        throw new WebApplicationException(new Exception(
-                "Couldn't find transformation for " + node.getPath()
-                        + " and transformation key " + key), SC_BAD_REQUEST);
+        final FedoraBinary transform = (FedoraBinary) transformResource.getChildren()
+                .filter(child -> rdfStringTypes.contains(child.getPath()))
+                .findFirst()
+                .orElseThrow(() -> new TransformNotFoundException(String.format("Couldn't find transformation for {} and transformation key {}", resource.getPath(), key)));
```

Comment:
During the extraction of fcrepo-transform from the core codebase the checkstyle plugin was dropped. Please trim this line to 120 characters or less in advance of that plugin being reintroduced.